test/extensions/io_socket/user_space/io_handle_impl_platform_test.cc

```diff
@@ -0,0 +1,62 @@
+#include "envoy/common/platform.h"
+#include "envoy/event/file_event.h"
+
+#include "extensions/io_socket/user_space/io_handle_impl.h"
+
+#include "test/mocks/event/mocks.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace IoSocket {
+namespace UserSpace {
+namespace {
+
+using testing::NiceMock;
+
+class MockFileEventCallback {
+public:
+  MOCK_METHOD(void, called, (uint32_t arg));
+};
+
+class IoHandleImplPlatformTest : public testing::Test {
+public:
+  IoHandleImplPlatformTest() {
+    std::tie(io_handle_, io_handle_peer_) = IoHandleFactory::createIoHandlePair();
+  }
+
+  ~IoHandleImplPlatformTest() override {
+    if (io_handle_->isOpen()) {
+      io_handle_->close();
+    }
+    if (io_handle_peer_->isOpen()) {
+      io_handle_peer_->close();
+    }
+  }
+
+  std::unique_ptr<IoHandleImpl> io_handle_;
+  std::unique_ptr<IoHandleImpl> io_handle_peer_;
+  NiceMock<Event::MockDispatcher> dispatcher_;
+  MockFileEventCallback cb_;
+};
+
+// Confirm that PlatformDefaultTriggerType is not level trigger type because IoHandle doesn't
+// support level trigger type. Specifically, windows platform user could use EmulatedEdge or
+// edge trigger type.
+TEST_F(IoHandleImplPlatformTest, CreatePlatformDefaultTriggerType) {
```

Comment:
Why is this test case in a separate file?  Please move to io_handle_impl_test.cc

Reply:
Yeah... it no longer fail on windows. Merged